gst/nnstreamer/include/nnstreamer_plugin_api_trainer.h

```diff
@@ -39,7 +39,8 @@ typedef struct _GstTensorTrainerProperties
   unsigned int num_validation_samples;    /**< The number of validation sample used to valid the model. */
   unsigned int num_epochs;    /**< The number of repetition of total training and validation sample. subplugin must receive total samples((num_training_samples + num_validation_samples) * num_epochs) */
 
-  GCond *training_complete_cond;    /**< Tensor trainer wait when receive EOS before model training is complete, subplugin should send signal when model training is complete. */
+  GCond *training_complete_cond;    /**< tensor_trainer wait when receive EOS before model training is complete, subplugin should send signal when model training is complete. */
+  GCond *epoch_complete_cond;    /**< One of epochs is complete, tensor_trainer wait for one of epochs to complete before getting the model stats from the subplugin. */
```

Comment:
I don't feel safe with this. Plugins providing GCond to main module?
It'd better let a plugin call an API for completion or an event. (i.e., let the main module provide an internal API, ```notify ( GstTensorTrainer *, EventType, EventData);```)

And you are not sharing an associated mutex.

Reply:
Thank you.